edx_proctoring/models.py

```diff
@@ -321,6 +322,53 @@ def get_active_student_attempts(self, user_id, course_id=None):
 
         return self.filter(filtered_query).order_by('-created')  # pylint: disable=no-member
 
+    def get_user_onboarding_attempt(self, user_id, course_id=None):
+        """
+        Filter user's onboarding attempts within the last two years, and return one of the
+        following:
+            1. The last verified onboarding attempt, if it exists
+            2. The most recent attempt, if there is no verified attempt
+            3. None, if there are no onboarding attempts
+        
+        Parameters:
+            * user_id: ID of the user
+            * course_id: Optional, used if we want to prioritize attempts from a certain course
+                         (I.E. for the instructor dashboard)
+        """
+        earliest_allowed_date = timezone.now() - timedelta(years=2)
```

Comment:
I'm not really sure of the best solution to this potential problem (maybe it's not even a problem), but I wanted to get my thoughts out there.

This method will run code that will hit the database multiple times in each call to it, say _m_. When you call bool on a queryset, the queryset is evaluated. If we're calling this function in a loop _n_  times for _n_ users we'd be making _m*n_ database calls. Could you take a look at the Django debug toolbar and see if you can see the database queries? Perhaps Django is smart enough to optimize to some degree. If not, it may make sense to think about how we can do this in fewer calls to the database if it seems like it might be problematic. But hopefully not!

Reply:
Thank you for pointing this out. I think taking advantage of [exists()](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#exists) will allow better performance (but by how much, I'm not sure yet)